rfcs/013-release_deployment_tracking/README.md

```diff
@@ -4,396 +4,460 @@
+- **deployment set:** A set of deployed **services** created from a **release set** that has been deployed into an **environment**.
 
-This mechanism will be provided via a python application, packaged in a docker container distributed via the https://github.com/wellcometrust/dockerfiles repository.
+#### How these terms fit together
 
-### Project structure
+![Terms](terms.png)
 
-> **Project:** A set of services that when composed perform a function.
+## What we do now
 
-In order to build releases that describe which version of a service to deploy to a particular environment we need a machine readable description of project structure.
+See the documentation on [version 1](v1/README.md).
 
-We propose the following structure:
+### Problems with the current approach
 
-```json
-{
-  "project": {
-    "id": "id",
-    "name": "project_name",
-    "profile": "aws_profile"
-  },
-  "environments" : [
-    {
-      "id": "environment_id",
-      "name": "environment_name"
-    }
-  ]
-}
+- It is not clear how to release a single service
+- In order to actually deploy something there are multiple steps:
+  - Create a release set using the CLI tool
+  - Deploy a release set  using the CLI tool
+  - Run `terraform apply` to _actually_ update the running services
+- Release/Deploy descriptions are not well used / hidden
+- Poor visibility of what is actually deployed
+
+#### Moving away from terraform for deployment
+
+We currently use `terraform apply` to deploy services at a particular release hash. The choice to use terraform was driven by a requirement to describe our task definitions in code. 
+
+Separating service deployment from infrastructure changes is desirable as infra/service deployments have differing concerns and pace, i.e. high-value infrequent (infra), vs. low-value frequent (deploying new versions of services). 
+
+Running terraform in a CI environment like Travis is not desirable as giving an automated environment the power to run infrastructure updates needs careful consideration.
+
+#### Why this is hard
+
+An ECS task definition contains configuration for volume mounts, CPU & memory requirements, as well as indicating the container image URI to use when creating tasks.
+
+The container image URI cannot be updated independently from other parameters in a task definition. This makes ignoring a change to the task definition difficult, see this [epic GitHub issue thread](https://github.com/terraform-providers/terraform-provider-aws/issues/632).
+
+When terraform updates a task definition it has a version of the task definition in code to send to ECS, the ECS Service is then updated by terraform to point at that new task definition and a deployment is started in ECS.
+
+However if the task definition is updated and differs from that recorded by the terraform state (which updating the image URI _would_ cause) terraform will attempt to return the task definition to a known state, which would be undesirable.
```

Comment:
I think I understand these sentences, but they're not so clear. Here's how I parsed it:

> When Terraform updates an ECS task definition, it sends the task definition to ECS and records that in the Terraform state. Terraform then tells the ECS service to use the new task definition, and ECS starts appropriate containers for the task.
>
> If the task definition is updated outside Terraform (including changing the image URI), then Terraform will attempt to revert the task definition to the version in its state. This could cause a service to go backwards.